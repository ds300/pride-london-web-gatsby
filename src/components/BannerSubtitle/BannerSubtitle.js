import styled from 'styled-components'

const BannerTitle = styled.h3`
  background-color: purple;
  display: inline-block;
  color: white;
  padding: 0 5px;
`

export default BannerTitle
