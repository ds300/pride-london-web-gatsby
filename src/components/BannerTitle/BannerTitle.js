import styled from 'styled-components'

const BannerTitle = styled.h1`
  background-color: grey;
  display: inline-block;
  color: purple;
  margin-bottom: 0;
  padding: 0 5px;

  &:after {
    height: 20px;
    display: inline-block;
  }
`

export default BannerTitle
