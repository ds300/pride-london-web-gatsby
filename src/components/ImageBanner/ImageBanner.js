import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import BannerTitle from '../BannerTitle/BannerTitle'
import BannerSubtitle from '../BannerSubtitle/BannerSubtitle'

const BackgroundImage = styled.div`
  justify-content: center;
  background-image: url(${props => props.imageSrc});
  background-size: cover;
  background-repeat: no-repeat;
  display: flex;
  flex-direction: column;
  padding-left: 15px;
  height: 400px;
`

const ImageBanner = ({ titleText, subtitleText, imageSrc }) => (
  <BackgroundImage imageSrc={imageSrc}>
    <BannerTitle>{titleText}</BannerTitle>
    <BannerSubtitle>{subtitleText}</BannerSubtitle>
  </BackgroundImage>
)

ImageBanner.propTypes = {
  imageSrc: PropTypes.string,
  subtitleText: PropTypes.string,
  titleText: PropTypes.string,
}

ImageBanner.defaultProps = {
  imageSrc: '',
  subtitleText: '',
  titleText: '',
}

export default ImageBanner
