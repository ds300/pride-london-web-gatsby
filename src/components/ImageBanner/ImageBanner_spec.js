import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow, mount } from 'enzyme'

import ImageBanner, { BackgroundImage } from './ImageBanner'
import BannerTitle from '../BannerTitle/BannerTitle'
import BannerSubtitle from '../BannerSubtitle/BannerSubtitle'

describe('ImageBanner', () => {
  it('should render', () => {
    const component = renderer.create(<ImageBanner />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should render a <BannerTitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find(BannerTitle).length).toBe(1)
  })

  it('should render a <BannerSubtitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find(BannerSubtitle).length).toBe(1)
  })

  it('should render the titleText from props to BannerTitle ', () => {
    const titleText = 'Here is a test title!'
    const wrapper = mount(<ImageBanner />)
    wrapper.setProps({ titleText })
    expect(wrapper.find(BannerTitle).text()).toBe(titleText)
  })

  it('should render the subtitleText from props to BannerTitle ', () => {
    const subtitleText = 'And here is a test subtitle!'
    const wrapper = mount(<ImageBanner />)
    wrapper.setProps({ subtitleText })
    expect(wrapper.find(BannerSubtitle).text()).toBe(subtitleText)
  })

  it('should pass the imageSrc prop to the BackgroundImage component', () => {
    const imageSrc = 'http://www.pics.com/coolpic'
    const wrapper = shallow(<ImageBanner />)
    wrapper.setProps({ imageSrc })
    expect(wrapper.find(BackgroundImage).props().imageSrc).toBe(imageSrc)
  })
})
