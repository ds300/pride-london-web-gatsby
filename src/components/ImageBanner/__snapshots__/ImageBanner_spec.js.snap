// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBanner should render 1`] = `
.c1 {
  background-color: grey;
  display: inline-block;
  color: purple;
  margin-bottom: 0;
  padding: 0 5px;
}

.c1:after {
  height: 20px;
  display: inline-block;
}

.c2 {
  background-color: purple;
  display: inline-block;
  color: white;
  padding: 0 5px;
}

.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-image: url();
  background-size: cover;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 15px;
  height: 400px;
}

<div
  className="c0"
>
  <div>
    <h1
      className="c1"
    >
      
    </h1>
    <br />
    <h3
      className="c2"
    >
      
    </h3>
  </div>
</div>
`;
